op-chain-ops/devkeys/devkeys.go

```diff
@@ -0,0 +1,116 @@
+package devkeys
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Domain separates the usage of development keys by broad kind of usage.
+// E.g. a general user account, a L1 thing, a superchain thing, a L2 thing, etc.
+type Domain uint64
+
+const (
+	// DefaultKeyDomain is the domain for keys that get pre-funded in every chain.
+	// A 0 chain ID should be used for this domain.
+	DefaultKeyDomain Domain = 0
+	// L1OperatorKeyDomain is the domain for keys used in L1 ops
+	L1OperatorKeyDomain Domain = 1
+	// L1UserKeyDomain is the domain for pre-funded user accounts specific to L1
+	L1UserKeyDomain Domain = 2
+	// SuperchainKeyDomain is the domain for superchain contracts.
+	// The L1 chain ID should be used as chain ID for this domain.
+	SuperchainKeyDomain Domain = 3
+	// L2OperatorKeyDomain is the domain for L2 operator keys
+	L2OperatorKeyDomain Domain = 4
+	// L2UserKeyDomain is the domain for pre-funded L2 user accounts specific to a single L2
+	L2UserKeyDomain Domain = 5
+)
+
+func (d Domain) String() string {
+	switch d {
+	case DefaultKeyDomain:
+		return "default"
+	case L1OperatorKeyDomain:
+		return "l1-operator"
+	case L1UserKeyDomain:
+		return "l1-user"
+	case SuperchainKeyDomain:
+		return "superchain"
+	case L2OperatorKeyDomain:
+		return "l2-operator"
+	case L2UserKeyDomain:
+		return "l2-user"
+	default:
+		return fmt.Sprintf("unknown-domain-%d", uint64(d))
+	}
+}
+
+// Role separates the usage of development keys by exact role, within a Domain.
+// Some roles may not be applicable to every Domain, those can be ignored.
+type Role uint64
+
+const (
+	DeployerRole                   Role = 0
+	ProposerRole                   Role = 1
+	BatcherRole                    Role = 2
+	SequencerP2PRole               Role = 3
+	ChallengerRole                 Role = 4
+	ProxyAdminOwnerRole            Role = 5
+	SuperchainConfigGuardianRole   Role = 6
+	FinalSystemOwnerRole           Role = 7
```

Comment:
We should rename `FinalSystemOwner` to be honest - its not clear what it means. iirc its the L1 `ProxyAdminOwner` and then `ProxyAdminOwner` is actually `L2ProxyAdminOwner`

Reply:
We also have `DependencySetManager` role introduced as part of interop, the standard config will have OP Gov be that role

Reply:
Updated, please have a look